pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1092,12 +1092,31 @@ func buildDestination(ctx RouteContext, to k8s.BackendRef, ns string,
 		}
 		inferencePoolServiceName, _ := InferencePoolServiceName(string(to.Name))
 		hostname := inferencePoolServiceName + "." + namespace + ".svc." + ctx.DomainSuffix
-		svc := ctx.LookupHostname(hostname, namespace)
-		if svc == nil {
+
+		// Look up shadow service directly from ctx.Services
+		// to ensure we get the latest shadow service even if PushContext hasn't been updated yet
```

Comment:
Wouldn't this be eventually consistent though? I'm not sure we need the extra fetch?

Reply:
You're right!!, The KRT dependency tracking automatically triggers HTTPRoute recomputation when the shadow service is created. The initial failures resolve within milliseconds through automatic reconciliation.